### PR TITLE
Add runner group configuration for self-hosted runners under GitHub organization 

### PR DESCRIPTION
**Pull request by yhaliaw**
Enables the charm to register self-hosted runners under a GitHub organization to custom runner groups.
Previously, the charm will register all spawned runners under the `default` runner group.

### CODE REVIEW

**Review by jdkandersson — DISMISSED** [review]

File: config.yaml [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
+    type: int
+    default: 2
+    description: >
+      The number of CPUs used per virtual machine runner.
   vm-memory:
     type: string
     default: 7GiB
     description: >
-        The memory size used per virtual machine runner.
+      The memory size used per virtual machine runner.
```
Comment (yanksyoon) [inline config.yaml:33]: Would this phrasing help be clear about the accepted units? I think sometimes users might put just a number of different units as the input :)
```suggestion
      Amount of memory to allocate per virtual machine runner. Positive integers with GiB suffix.
```
Taken from [multipass documentation](https://multipass.run/docs/launch-command) :D

File: config.yaml [hunk 2]
```diff
@@ -1,45 +1,50 @@
...
   vm-disk:
     type: string
     default: 10GiB
     description: >
-        The root disk size used per virtual machine runner.
+      The root disk size used per virtual machine runner.
```
Comment (yanksyoon) [inline config.yaml:38]:
```suggestion
      Amount of disk space to allocate to root disk for virtual machine runner. Positive integers with GiB suffix.
```
Same as above :)

File: config.yaml [hunk 3]
```diff
@@ -1,45 +1,50 @@
...
-      type: string
-      default: ""
-      description: >
-        The path of the GitHub repository to register the self-hosted runners. Either an
-        organization name or repository in 'org/repo' format.
+    type: string
+    default: ""
+    description: >
+      The path of the GitHub repository to register the self-hosted runners. Either an
+      organization name or repository in 'org/repo' format.
```
Comment (yanksyoon) [inline config.yaml:7]: Would describing the differences between 'org' vs 'org/repo' be helpful? :)

File: src/runner.py [hunk 4]
```diff
@@ -424,23 +424,28 @@ def _register_runner(self, registration_token: str, labels: Sequence[str]) -> No
...
 
+        register_cmd = [
+            "/usr/bin/sudo",
+            "-u",
+            "ubuntu",
+            str(self.config_script),
+            "--url",
+            f"https://github.com/{self.config.path.path()}",
+            "--token",
+            f"{registration_token}",
```
Comment (yanksyoon) [inline src/runner.py:435]: I think the f string here could be removed :)

File: src/runner.py [hunk 5]
```diff
@@ -424,23 +424,28 @@ def _register_runner(self, registration_token: str, labels: Sequence[str]) -> No
...
+            "--ephemeral",
+            "--name",
+            f"{self.instance.name}",
```
Comment (yanksyoon) [inline src/runner.py:438]: i think the f-string here could be removed :)

**Review by yanksyoon — DISMISSED** [review]
LGTM! Minor changes requested :D

**Review by jdkandersson — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Pull request merged**